components/widgets/eureka/health-status.js

```diff
@@ -0,0 +1,163 @@
+import { Component } from 'react'
+import fetch from 'isomorphic-unfetch'
+import yup from 'yup'
+import Widget from '../../widget'
+import Table, { Th, Td } from '../../table'
+import { basicAuthHeader } from '../../../lib/auth'
+import styled from 'styled-components'
+
+const schema = yup.object().shape({
+  url: yup.string().url().required(),
+  interval: yup.number(),
+  baseQuery: yup.string().required(),
+  appsQuery: yup.string().required(),
+  healthQuery: yup.string().required(),
+  title: yup.string(),
+  minimumInstances: yup.number(),
+  appNamePattern: yup.string(),
+  authKey: yup.string()
+})
+
+const EurekaDiv = styled.div`
+  background-color: ${props => props.hasError ? props.theme.palette.errorColor : props.theme.palette.canvasColor};
+`
+
+export default class EurekaHealthStatus extends Component {
+  static defaultProps = {
+    interval: 1000 * 60 * 60,
+    title: 'Eureka Health Status',
+    minimumInstances: 2,
+    appNamePattern: ''
+  }
+
+  state = {
+    error: false,
+    loading: true,
+    appStatus: '',
+    infoMessage: ''
+  }
+
+  componentDidMount () {
+    schema.validate(this.props)
+    .then(() => this.fetchInformation())
+    .catch((err) => {
+      console.error(`${err.name} @ ${this.constructor.name}`, err.errors)
+      this.setState({ error: true, loading: false })
+    })
+  }
+
+  componentWillUnmount () {
+    clearInterval(this.interval)
+  }
+
+  checkInstanceCount (minimumInstances, appNamePattern, appList) {
+    let hasError = false
+    appList.forEach(function (entry) {
+      if (entry.name.startsWith(appNamePattern) && entry.instance.length < minimumInstances) {
+        hasError = true
+      }
+    })
+    return hasError
+  }
+
+  async checkInstanceHealth (url, appNamePattern, appList) {

```

Comment:
I think we can optimize this function in the next step. @chrishelgert Can you support @michl-b, please?